include/aspect/simulator_access.h

```diff
@@ -734,6 +739,15 @@ namespace aspect
       const NewtonHandler<dim> &
       get_newton_handler () const;
 
+#ifdef ASPECT_USE_WORLD_BUILDER
+      /**
+       * Return a reference to the world builder that controls the setup of
+       * initial conditions.
+       */
+      const WorldBuilder::World &
+      get_world_builder () const;
+#endif
```

Comment:
I think this is the wrong solution. We try to be compatible by providing the same interface (if we can) whether or not we have an external library. My preference would be to *always* declare the function, and if there is no WorldBuilder available, just throw an exception.

Reply:
See #2614.